omod/src/main/java/org/openmrs/module/fhir2/web/filter/AuthenticationFilter.java

```diff
@@ -54,13 +54,18 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 							Context.authenticate(userAndPass[0], userAndPass[1]);
 						}
 						catch (Exception ignored) {
-							HttpServletResponse httpResponse = (HttpServletResponse) response;
-							httpResponse.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Not authenticated");
+							if (!response.isCommitted()) {
+								HttpServletResponse httpResponse = (HttpServletResponse) response;
+								httpResponse.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Not authenticated");
+							}
+							return;
 						}
```

Comment:
@ibacher The only error that is left now is this, even on giving the correct password and username for basic auth the catch block is executed with `invalid credentials error`

Reply:
@VaishSiddharth This change should be made outside of this PR, since it's outside of the scope of the SMART stuff.